pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -8,76 +8,215 @@ import (
+	dcInformer.AddEventHandler(framework.ResourceEventHandlerFuncs{
+		AddFunc:    c.addDeploymentConfig,
+		UpdateFunc: c.updateDeploymentConfig,
+		DeleteFunc: c.deleteDeploymentConfig,
+	})
+
+	// Use a shared rc controller to verify that the rc store is synced.
+	// Also add rc event handlers that will trigger dc requeues.
+	c.rcController = rcInformer.GetController()
+	c.rcStore.Indexer = rcInformer.GetIndexer()
+	rcInformer.AddEventHandler(framework.ResourceEventHandlerFuncs{
+		AddFunc:    c.addReplicationController,
+		UpdateFunc: c.updateReplicationController,
+		DeleteFunc: c.deleteReplicationController,
+	})
+
+	// Use a shared pod controller to verify if the pod store is synced.
+	c.podController = podInformer.GetController()
+
+	return c
 }
 
-// Create creates a DeploymentConfigController.
-func (factory *DeploymentConfigControllerFactory) Create() controller.RunnableController {
-	deploymentConfigLW := &cache.ListWatch{
-		ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
-			return factory.Client.DeploymentConfigs(kapi.NamespaceAll).List(options)
-		},
-		WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
-			return factory.Client.DeploymentConfigs(kapi.NamespaceAll).Watch(options)
-		},
+// Run begins watching and syncing.
+func (c *DeploymentConfigController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
 	}
-	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
-	cache.NewReflector(deploymentConfigLW, &deployapi.DeploymentConfig{}, queue, 2*time.Minute).Run()
+	<-stopCh
+	glog.Infof("Shutting down deploymentconfig controller")
+	c.queue.ShutDown()
+}
 
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartRecordingToSink(factory.KubeClient.Events(""))
-	recorder := eventBroadcaster.NewRecorder(kapi.EventSource{Component: "deploymentconfig-controller"})
+func (c *DeploymentConfigController) addDeploymentConfig(obj interface{}) {
+	dc := obj.(*deployapi.DeploymentConfig)
+	glog.V(4).Infof("Adding deployment config %q", dc.Name)
+	c.enqueueDeploymentConfig(dc)
+}
+
+func (c *DeploymentConfigController) updateDeploymentConfig(old, cur interface{}) {
+	oldDc := old.(*deployapi.DeploymentConfig)
+	glog.V(4).Infof("Updating deployment config %q", oldDc.Name)
+	c.enqueueDeploymentConfig(cur.(*deployapi.DeploymentConfig))
+}
+
+func (c *DeploymentConfigController) deleteDeploymentConfig(obj interface{}) {
+	dc, ok := obj.(*deployapi.DeploymentConfig)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			glog.Errorf("Couldn't get object from tombstone %+v", obj)
+			return
+		}
+		dc, ok = tombstone.Obj.(*deployapi.DeploymentConfig)
+		if !ok {
+			glog.Errorf("Tombstone contained object that is not a deployment config: %+v", obj)
+			return
+		}
+	}
+	glog.V(4).Infof("Deleting deployment config %q", dc.Name)
```

Comment:
seems like you should do this only if `ok`.  reverse the condition above to reduce nesting?  Also, you can drop through to here from your if above.